src/dune_cache_storage/layout.mli

```diff
@@ -12,8 +12,11 @@ open Import
 val root_dir : Path.t
 
 (** Create a few subdirectories in [root_dir]. We expose this function because
-    we don't want to modify the file system when the cache is disabled. *)
-val create_cache_directories : unit -> unit
+    we don't want to modify the file system when the cache is disabled.
+
+    Returns whether creation has succeeded or in the case of error which directory
+    could not be created. *)
+val create_cache_directories : unit -> (unit, Path.t) result
```

Comment:
I think the error should include the underlying unix error so that you can actually print what went wrong.